src/features/offer/components/SameArtistPlaylist/api/fetchOffersByArtist.ts

```diff
@@ -0,0 +1,43 @@
+import { captureAlgoliaError } from 'libs/algolia/fetchAlgolia/AlgoliaError'
+import { offerAttributesToRetrieve } from 'libs/algolia/fetchAlgolia/buildAlgoliaParameters/offerAttributesToRetrieve'
+import { client } from 'libs/algolia/fetchAlgolia/clients'
+import { env } from 'libs/environment'
+import { HitOffer, Offer } from 'shared/offer/types'
+
+export type FetchOfferByArtist = {
+  artist: string | null | undefined
+  ean: string | null | undefined
+}
+
+export type HitOfferWithArtistAndEan = Offer & {
+  offer: HitOffer & {
+    artist: string
+    ean: string
+  }
+}
+
+export const fetchOffersByArtist = async ({
+  artist,
+  ean,
+}: FetchOfferByArtist): Promise<HitOfferWithArtistAndEan[]> => {
+  const index = client.initIndex(env.ALGOLIA_OFFERS_INDEX_NAME)
+
+  if (!artist || !ean) return []
+
+  const getFirstArtist = artist?.split(' ; ')[0]
```

Comment:
Ce n'est pas un `getter`, tu peux juste l'appeler 
```suggestion
  const firstArtist = artist?.split(' ; ')[0]
```